src/Program/Model/City.hs

```diff
@@ -344,6 +346,10 @@ foreign import javascript unsafe "$r = $3.slice(); $r[$1] = $2;"
 isEmptyCity :: City -> Bool
 isEmptyCity c = collectionLength (objectsIn c) == 0
 
+-- | Filter CityObject properties according to hiddenProperties settings
+shownProps :: City -> CityObject -> JSVal
+shownProps ci obj = Prelude.foldr id (allProps obj) $ deleteProp <$> (hiddenProperties $ cityProperties ci)
```

Comment:
From your algorithm, you do JSON parse and stringify for each property in the list! This is too much.
1) I think it is better to not introduce function deleteProp at all
2) A better way for future improvement of qua-view, I think, is to change shownProps to return list, and do a proper list difference using Data.List  operations.